tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -464,7 +466,7 @@ def _assert_pantsd_keyboardinterrupt_signal(self, signum):
         config,
       )
 
-      checker.assert_started(timeout=12)
+      self._wait_for_waiter_run(checker)
```

Comment:
I'm confused... Wasn't the point of bumping the `assert_started` default timeout to `12` that you wouldn't need to explicitly lengthen it anywhere else?

I'm expecting zero calls to `assert_started` to override the timeout argument unless there's something specific in the test case that would make it be slower than normal (e.g. working against a giant source tree).

Can we get to a place where there's one value for timeout, which is 12 seconds or 16 seconds or whatever?

Reply:
16 seconds has consistently been necessary for the keyboard interrupt testing, and looking at the rest of this file I realize that the one place which happened to be calling `time.sleep(15)` also happened to be invoking the same `...:waiter` target. 16 seconds hasn't failed yet, so let's centralize to that and remove any overrides of the `timeout` argument.

Reply:
And for the record, I suspect that picking up more source files and traversing the fs is exactly why these calls require more time, further motivating #7372 so that we decouple testing how long it takes pantsd to start up from using it in this testing, because by doing both here we also slightly fail at both due to flakiness.

Reply:
Filed https://github.com/pantsbuild/pants/issues/7375 to cover how I've been thinking about addressing the startup times :)